content/post/2019-10-15.Rmd

```diff
@@ -0,0 +1,388 @@
+
+
+```{r, eval = FALSE}
+cppFunction('DataFrame createTrajectory(int n, double x0, double y0,
+            double a1, double a2, double a3, double a4, double a5,
+            double a6, double a7, double a8, double a9, double a10,
+            double a11, double a12, double a13, double a14) {
+            // create the columns
+            NumericVector x(n);
+            NumericVector y(n);
+            x[0]=x0;
+            y[0]=y0;
+            for(int i = 1; i < n; ++i) {
+            x[i] = a1+a2*x[i-1]+ a3*y[i-1]+ a4*pow(fabs(x[i-1]), a5)+ a6*pow(fabs(y[i-1]), a7);
+            y[i] = a8+a9*x[i-1]+ a10*y[i-1]+ a11*pow(fabs(x[i-1]), a12)+ a13*pow(fabs(y[i-1]), a14);
+            }
+            // return a new data frame
+            return DataFrame::create(_["x"]= x, _["y"]= y);
+            }
+            ')
+a1 <- -0.8
+a2 <- 0.4
+a3 <- -1.1
+a4 <- 0.5
+a5 <- -0.6
+a6 <- -0.1
+a7 <- -0.5
+a8 <- 0.8
+a9 <- 1.0
+a10 <- -0.3
+a11 <- -0.6
+a12 <- -0.3
+a13 <- -1.2
+a14 <- -0.3
+
+df <- createTrajectory(10000000, 1, 1, a1, a2, a3, a4, a5, a6,
+                       a7, a8, a9, a10, a11, a12, a13, a14)
+
+ggplot(df) +
+  geom_point(aes(x = x, y = y), shape = 46, alpha = .01, color = "black") +
+  coord_fixed() +
+  theme_void() -> plot
+
+ggsave("strange3.png", plot, height = 5, width = 5, units = 'in')
+
+```
+
+
+<center>
+![](../../static/post/2019-10-15_files/strange3.png){width=450px}
+</center>
+
+Ops! What happened? Why our image is so small now? The reason is because some few points are located quite far from the vast majority (you can think about them as *outliers*). Let's remove 5% of extreme points from both tails independently:
+
+```{r, eval = FALSE}
+mx <- quantile(df$x, probs = 0.05)
+Mx <- quantile(df$x, probs = 0.95)
+my <- quantile(df$y, probs = 0.05)
+My <- quantile(df$y, probs = 0.95)
+
+df %>% filter(x > mx, x < Mx, y > my, y < My) -> df
+
+ggplot(df) +
+  geom_point(aes(x = x, y = y), shape = 46, alpha = .01, color = "black") +
+  coord_fixed() +
+  theme_void() -> plot
+
+ggsave("strange3.png", plot, height = 5, width = 5, units = 'in')
+
+```
+
+
+<center>
+![](../../static/post/2019-10-15_files/strange5.png){width=450px}
+</center>
+
+Now a stunning image appears. 
+
+Sometimes you will need to process the data frame after you generate it since:
+
++ Extreme values can hide the pattern as happened before or even *scape* to infinite (solution: remove outliers).
++ Some image can be quite *boring*: from my point of view, the more *distributed* ae points on the XY plane, the nicer is the resulting image (solution: try with a small data frame and keep it if some measure of dispersion over the plane or the individual axis is high enough; in this case, compute the big picture).
+
+To end, let's change the appearance of our plot. We will use the `theme` function to customize it setting the background color to black and adding a white line as frame. 
```

Comment:
1/ "To end" es una expresión que nunca la he visto usar en ese contexto. Yo pondría "To conclude" o "To finish". Sin embargo, si tú las has visto usar y crees/sabes que es correcta, no tienes que cambiarla.